elasticapm/transport/base.py

```diff
@@ -135,7 +152,10 @@ def close(self):
         Cleans up resources and closes connection
         :return:
         """
-        self.flush(sync=True, start_flush_timer=False)
+        self._closed = True
+        self.queue("close", None)
+        if self._event_process_thread.is_alive():
+            self._event_process_thread.join()
```

Comment:
Unless you ensure the queue above didn't drop, this could block forever. If you do add a timeout to `close`, I think you would need to measure how long the `queue` call takes, and subtract that from the timeout that you then pass to `join`.